app/src/androidTest/java/com/github/sdp_begreen/begreen/activities/MainActivityTest.kt

```diff
@@ -154,4 +174,146 @@ class MainActivityTest {
         onView(withId(R.id.settingsFragment))
             .check(matches(isDisplayed()))
     }
+
+    // TODO All the following test requiring a comparison of bitmap have been disabled for now
+    // TODO Need to find a better way to compare bitmap image
+
+    @Test
+    fun correctInfoDisplayedForAuthenticatedUser() {
+
+        runBlocking {
+            Firebase.auth.signOut()
+            Firebase.auth.signInWithEmailAndPassword("user1@email.ch", "123456").await()
+
+        }
+
+        onView(withId(R.id.bottomMenuUser))
+            .check(matches(isDisplayed()))
+            .perform(click())
+
+        onView(withId(R.id.nav_drawer_username_textview))
+            .check(matches(withText("User Test 1")))
+
+        onView(withId(R.id.nav_drawer_description_textview))
+            .check(matches(withText("That's the awesome description of test user 1")))
+
+        /*activityRule.scenario.onActivity {
+            val drawable: BitmapDrawable =
+                it.findViewById<ImageView>(R.id.nav_drawer_profile_picture_imageview).drawable as BitmapDrawable
+
+            val expectedBitmap = BitmapFactory.decodeResource(it.resources, R.drawable.marguerite_test_image)
+            assertThat(drawable.bitmap, equalsBitmap(expectedBitmap))
+        }*/
+    }
+
+    @Test
+    fun defaultValueDisplayedForUnauthenticatedUser() {
+        Firebase.auth.signOut() // Ensure signed out
+
+        assertThat(Firebase.auth.currentUser, nullValue())
+
+        onView(withId(R.id.bottomMenuUser))
+            .check(matches(isDisplayed()))
+            .perform(click())
+
+        onView(withId(R.id.nav_drawer_username_textview))
+            .check(matches(withText("Username")))
+
+        onView(withId(R.id.nav_drawer_description_textview))
+            .check(matches((withText("More Info on user"))))
+
+        /*activityRule.scenario.onActivity {
+            val drawable: BitmapDrawable =
+                it.findViewById<ImageView>(R.id.nav_drawer_profile_picture_imageview).drawable as BitmapDrawable
+
+            val expectedBitmap = BitmapFactory.decodeResource(it.resources, R.drawable.blank_profile_picture)
+            assertThat(drawable.bitmap, equalsBitmap(expectedBitmap))
+        }*/
+
+    }
+
+    @Test
+    fun defaultValueDisplayedForAuthenticatedUserNotInDB() {
+        runBlocking {
+            Firebase.auth.signOut()
+            Firebase.auth.signInWithEmailAndPassword("notInDb@email.com", "123456").await()
```

Comment:
It's a bit weird that this user _is_ in fact in the DB, per the data dump also added with this PR. I guess "not in DB" has a meaning that is not quite "not known to Firebase" ?

Reply:
Yes I agree with you the name is maybe not the best. But what I wanted to represent with this `notInDB` user is a user that can connect, but for whom we don't have an entry in the `RealTimeDatabase` so the default value should be displayed.

But I agree I should have name it `NotInRealTimeDb` or something.